packages/explorer/src/app/(explorer)/api/utils/decodeTable.ts

```diff
@@ -0,0 +1,48 @@
+import { Hex, decodeAbiParameters, parseAbiParameters } from "viem";
+import { hexToResource } from "@latticexyz/common";
+import { Schema, Table } from "@latticexyz/config";
+import { getSchemaPrimitives, hexToSchema } from "@latticexyz/protocol-parser/internal";
+
+export type DeployedTable = Table & {
+  valueSchema: Schema;
+  keySchema: Schema;
+};
```

Comment:
where do we use `keySchema` and `valueSchema` in explorer? curious if we can move away from it or if it's important to have

Reply:
ah okay, so it just sounds like the helpers we have don't cover the use cases here to avoid having to pass around keySchema/valueSchema

we should add those to protocol-parser etc. but doesn't have to happen here

Reply:
just checked, look like it does have some useful utilities like `getKey` or `getKeyTuple`. I'll check more which ones I can use

Reply:
Alright, did some refactoring, and made use of `getValueSchema()` + `getKeySchema()`. So removed `DeployedTable` type, and just using `Table` from `@latticexyz/config`. 

A nice utility that would be nice to have and that I can't find is constructing the key field for `setField` method. Not sure though if I just can't find it.